theseus/embodied/robot/joint.py

```diff
@@ -8,23 +8,24 @@
 from typing import Optional
 import torch
 
+from .link import Link
 from theseus.labs.lie_functional import so3, se3
 
 
 class Joint(abc.ABC):
     def __init__(
         self,
         name: str,
-        parent: int = -1,
-        child: int = -1,
+        id: int = -1,
```

Comment:
I think making `id` optional with default to `None` is cleaner that using `-1`. Rather than checking `id >= 0`, we can check `id is not None`. It is also easier to understand when others read the code.

Reply:
Good idea.